### PR TITLE
Add a callback handler for Context (https://getcontext.ai)

### PR DESCRIPTION
**Pull request by agamble**
### Description

Adding a callback handler for Context. Context is a product analytics platform for AI chat experiences to help you understand how users are interacting with your product.

I've added the callback library + an example notebook showing its use.

### Dependencies

Requires the user to install the `context-python` library. The library is lazily-loaded when the callback is instantiated.

### Announcing the feature

We spoke with Harrison a few weeks ago about also doing a blog post announcing our integration, so will coordinate this with him. Our Twitter handle for the company is @getcontextai, and the founders are @_agamble and @HenrySG.

Tagging @agola11 for callbacks.

Thanks in advance!

### CODE REVIEW

File: langchain/callbacks/context_callback.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        (
+            self.context,
+            self.credential,
+            self.conversation_model,
+            self.message_model,
+            self.message_role_model,
+            self.rating_model,
+        ) = import_context()
+
+        self.client = self.context.ContextAPI(credential=self.credential(token))
```
Comment (baskaryan) [inline langchain/callbacks/context_callback.py:97]: up to you if you want to do this, in other places we usually provide an option to specify api tokens via env vars. could do something like
```python
import os

def __init__(self, token: Optional[str] = None,...):
  ...
  token = token or os.environ.get("CONTEXT_API_TOKEN")
  ...

File: langchain/callbacks/context_callback.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+        self.chain_run_id = None
+
+        self.within_chat_model = False
+        self.llm_model = None
```
Comment (baskaryan) [inline langchain/callbacks/context_callback.py:102]: maybe im just not seeing it, but is this used anywhere?

**Review by baskaryan — APPROVED** [review]
two small things but overall lgtm!

File: langchain/callbacks/context_callback.py (hunk 1 above)
Comment (agamble) [reply langchain/callbacks/context_callback.py:102]: Nice idea, have added.

**Pull request merged**